plopfile.mjs

```diff
@@ -0,0 +1,66 @@
+export default function (/** @type {import('plop').NodePlopAPI} */ plop) {
+  plop.setGenerator("new-stencil-component", {
+    description: "Create a new Stencil Component",
+    prompts: [
+      {
+        type: "input",
+        name: "name",
+        message: "What is your component name?",
+      },
+    ],
+    actions: [
+      {
+        type: "add",
+        templateFile: "plop-templates/new-stencil-component/stencil-component.tsx.hbs",
+        path: "packages/core/src/components/{{ kebabCase name }}/{{ kebabCase name }}.tsx",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/new-stencil-component/stencil-component.interfaces.ts.hbs",
+        path: "packages/core/src/components/{{ kebabCase name }}/{{ kebabCase name }}.interfaces.ts",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/new-stencil-component/stencil-component.scss.hbs",
+        path: "packages/core/src/components/{{ kebabCase name }}/{{ kebabCase name }}.scss",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/new-stencil-component/readme.md.hbs",
+        path: "packages/core/src/components/{{ kebabCase name }}/readme.md",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/storybook-boilerplate/component.args.ts.hbs",
+        path: "packages/dso-toolkit/src/components/{{ kebabCase name }}/{{ kebabCase name }}.args.ts",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/storybook-boilerplate/component.models.ts.hbs",
+        path: "packages/dso-toolkit/src/components/{{ kebabCase name }}/{{ kebabCase name }}.models.ts",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/storybook-boilerplate/component.stories-of.ts.hbs",
+        path: "packages/dso-toolkit/src/components/{{ kebabCase name }}/{{ kebabCase name }}.stories-of.ts",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/storybook-boilerplate/index.ts.hbs",
+        path: "packages/dso-toolkit/src/components/{{ kebabCase name }}/index.ts",
+      },
+      // modify packages/dso-toolkit/src/index.ts with exports from new component
+      {
+        type: "add",
+        templateFile: "plop-templates/storybook-boilerplate/component.core-stories.ts.hbs",
+        path: "storybook/src/components/{{ kebabCase name }}/{{ kebabCase name }}.core-stories.ts",
+      },
+      {
+        type: "add",
+        templateFile: "plop-templates/storybook-boilerplate/component.core-template.ts.hbs",
+        path: "storybook/src/components/{{ kebabCase name }}/{{ kebabCase name }}.core-template.ts",
+      },
+      // modify storybook/src/templates.ts with new component
```

Comment:
Nog wel een interessant uitdaging om via een customAction of append dit te realiseren